python/langsmith/client.py

```diff
@@ -3808,6 +3808,77 @@ def create_presigned_feedback_token(
         ls_utils.raise_for_status_with_text(response)
         return ls_schemas.FeedbackIngestToken(**response.json())
 
+    def create_presigned_feedback_tokens(
+        self,
+        run_id: ID_TYPE,
+        feedback_keys: Sequence[str],
+        *,
+        expiration: Optional[datetime.datetime | datetime.timedelta] = None,
+        feedback_configs: Optional[
+            Sequence[Optional[ls_schemas.FeedbackConfig]]
+        ] = None,
+    ) -> Sequence[ls_schemas.FeedbackIngestToken]:
+        """Create a pre-signed URL to send feedback data to.
+
+        This is useful for giving browser-based clients a way to upload
+        feedback data directly to LangSmith without accessing the
+        API key.
+
+        Args:
+            run_id:
+            feedback_key:
+            expiration: The expiration time of the pre-signed URL.
+                Either a datetime or a timedelta offset from now.
+                Default to 3 hours.
+            feedback_config: FeedbackConfig or None.
+                If creating a feedback_key for the first time,
+                this defines how the metric should be interpreted,
+                such as a continuous score (w/ optional bounds),
+                or distribution over categorical values.
+
+        Returns:
+            The pre-signed URL for uploading feedback data.
+        """
+        if feedback_configs is not None and len(feedback_keys) != len(feedback_configs):
+            raise ValueError(
+                "The length of feedback_keys and feedback_configs must be the same."
+            )
+        if not feedback_configs:
+            feedback_configs = [None] * len(feedback_keys)
+        body: List[Dict[str, Any]] = [
+            {
+                "run_id": run_id,
+                "feedback_key": feedback_key,
+                "feedback_config": feedback_config,
+            }
+            for feedback_key, feedback_config in zip(feedback_keys, feedback_configs)
+        ]
+        for part in body:
+            if expiration is None:
```

Comment:
ooc, why not make this once and re-use in the loop?

Reply:
because I wasn't thinking

Reply:
Updated